src/incubator/WheelPicker/index.tsx

```diff
@@ -39,47 +46,104 @@ export interface WheelPickerProps {
   const scrollView = useRef<Animated.ScrollView>();
   const [offset] = useValues([0], []);
+  let scrollOffset = useRef<number>(0).current;
   const onScroll = onScrollEvent({y: offset});
-  
-  const listSize = items?.length || 0;
-  const middleIndex = useMiddleIndex({itemHeight, listSize});
+
+  const {height, items, shouldControlComponent, index: currentIndex, getRowItemAtOffset} = usePresenter({
+    selectedValue,
+    items: propItems,
+    children,
+    itemHeight,
+    preferredNumVisibleRows: numberOfVisibleRows
+  });
+
+  useEffect(() => {
+    controlComponent();
+  });
+
+  /**
+   * The picker is a controlled component. This means we expect the
+   * to relay on `selectedValue` prop to be our
+   * source of truth - not the picker current value.
+   * This way, you can control disallow or mutate selection of some values.
+   */
+  const controlComponent = () => {
+    if (shouldControlComponent(scrollOffset)) {
+      scrollToIndex(currentIndex, true);
+    }
+  };
+
+  const scrollToPassedIndex = (animated: boolean = false) => {
+    scrollToIndex(currentIndex, animated);
+  };
+
+  const scrollToIndex = (index: number, animated: boolean) => {
+    if (scrollView.current?.getNode()) {
+      //@ts-ignore for some reason scrollToOffset isn't recognized
+      scrollView.current?.getNode()?.scrollToOffset({offset: index * itemHeight, animated: animated});
+    }
+  };
 
   const selectItem = useCallback(
     index => {
-      if (scrollView.current?.getNode()) {
-        //@ts-ignore for some reason scrollToOffset isn't recognized
-        scrollView.current?.getNode()?.scrollToOffset({offset: index * itemHeight, animated: true})
-      }
+      scrollToIndex(index, true);
     },
     [itemHeight]
   );
 
-  const onChange = useCallback((event: NativeSyntheticEvent<NativeScrollEvent>) => {
-    const index = middleIndex(event.nativeEvent.contentOffset.y);
-    onChangeEvent(index, items?.[index]);
-  }, []);
-  
-  const renderItem = useCallback(({item, index}) => {
-    return (
-      <Item
-        index={index}
-        itemHeight={itemHeight}
-        offset={offset}
-        activeColor={activeTextColor}
-        inactiveColor={inactiveTextColor}
-        style={textStyle}
-        {...item}
-        onSelect={selectItem}
-      />
-    );
-  }, [itemHeight]);
+  const onChange = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    scrollOffset = event.nativeEvent.contentOffset.y;
+    
+    const {index, value} = getRowItemAtOffset(scrollOffset);
+    onValueChange?.(value, index);
+  };
+
+  const renderItem = useCallback(
+    ({item, index}) => {
+      return (
+        <Item
+          index={index}
+          itemHeight={itemHeight}
+          offset={offset}
+          activeColor={activeTextColor}
+          inactiveColor={inactiveTextColor}
+          style={textStyle}
+          {...item}
+          onSelect={selectItem}
+        />
+      );
+    },
+    [itemHeight]
+  );
 
   const fader = useMemo(
```

Comment:
Is there any benefit from using `useMemo` here (for `fader`)? 
Why not just render `<Fader....` twice at the bottom

Reply:
I think Android could slow down UI thread sometime, while render fade / gradient views (also in native), so I thought to memo it even though it's not logically necessary.
but I removed from `separators` since it's just a simple view.

WDYT?